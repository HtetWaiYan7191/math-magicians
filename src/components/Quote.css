.quote-container {
  width: 50%;
  margin: 0 auto;
  padding: 10px;
  display: flex;
  flex-direction: column;
  border: 1px solid black;
  text-align: center;
}

.middle {
  text-align: center;
  width: 20%;
  margin: 0 auto;
}

.quote-container span {
  margin: 20px 0;
  font-weight: 600;
}

.quote {
  line-height: 27px;
}
