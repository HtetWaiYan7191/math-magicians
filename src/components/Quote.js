import React, { useEffect, useState } from 'react';
import './Quote.css';
import { ImSpinner10 } from 'react-icons/im';

function Quote() {
  const [data, setData] = useState([{ quote: '', author: '', category: '' }]);
  const [loading, setLoading] = useState(false);
  const [hasError, setHasError] = useState(false);

  useEffect(() => {
    const fetchData = async () => {
      const category = 'happiness';
      const url = `https://api.api-ninjas.com/v1/quotes?category=${category}`;
      const apiKey = 'h4lyJHiRHcwWlzfvQjccBw==vq4qYlL6Iybi4fRC';
      setLoading(true);
      try {
        const api = await fetch(url, {
          headers: {
            'X-Api-Key': apiKey,
          },
          contentType: 'application/json',
        });
        const result = await api.json();
        setData(result[0]);
      } catch (error) {
        setHasError(true);
      }
      setLoading(false);
    };
    fetchData();
  }, []);

  if (hasError) return <div className="middle">Something went wrong!</div>;

  if (loading) return <div className="middle"><ImSpinner10 className="rotate" /></div>;
  return (
    <div className="hero">
      <div className="quote-container">
        <div className="quote">{data.quote}</div>
        <span>
          By
          {' '}
          {' '}
          {data.author}
        </span>
      </div>
    </div>

  );
}

export default Quote;
