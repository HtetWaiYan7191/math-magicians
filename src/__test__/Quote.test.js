import React from 'react';
import renderer from 'react-test-renderer';
import Quote from '../components/Quote';

describe('Quote', () => {
  test('should match the snapShot', () => {
    const component = renderer.create(<Quote />);
    const instance = component.toJSON();
    expect(instance).toMatchSnapshot();
  });
});
