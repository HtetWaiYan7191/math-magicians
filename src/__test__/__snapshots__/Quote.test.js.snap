// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote should match the snapShot 1`] = `
<div
  className="hero"
>
  <div
    className="quote-container"
  >
    <div
      className="quote"
    />
    <span>
      By
       
       
    </span>
  </div>
</div>
`;
